fdbserver/include/fdbserver/DataDistributionTeam.h

```diff
@@ -22,6 +22,35 @@
 
 #include "fdbclient/StorageServerInterface.h"
 
+struct GetTeamRequest;
+namespace data_distribution {
+class EligibilityCounter {
+public:
+	// The type value are used to do bit operations to get combined type. Ex. combineType = LOW_CPU | LOW_DISK_UTIL .
+	// When adding more types, the value should be 2^n, like 4, 8, 16...
+	enum Type {
+		NONE = 0, // don't care about eligibility
+		LOW_CPU = 1,
+		LOW_DISK_UTIL = 2
+	};
+
+	// set the count of type to 0
+	void reset(Type type);
+
+	// return the minimal count of a combined eligible type
+	unsigned getCount(int combinedType) const;
+
+	// increase the count of type
+	void increase(Type type);
+
+	static int fromGetTeamRequest(GetTeamRequest const&);
+
+private:
+	std::map<Type, unsigned int> type_count;
```

Comment:
Assuming that we don't need to maintain order, `std::unordered_map` is preferable as it has better performance than `std::map`.